icu4c/source/tools/toolutil/ucm.cpp

```diff
@@ -354,7 +354,7 @@ checkBaseExtUnicode(UCMStates *baseStates, UCMTable *base, UCMTable *ext,
         /* compare the base and extension mappings */
         cmp=compareUnicode(base, mb, ext, me);
         if(cmp<0) {
-            if(intersectBase && (intersectBase!=2 || mb->bLen>1)) {
```

Comment:
What was going on here? Did someone expect an UBool to be 2?

Reply:
I don't remember the details of this code from somewhere around 2002... but I would be very surprised that I wrote this when intersectBase can never be 2.
It would be safer to change the type to int8_t.

Reply:
> [...] but I would be very surprised that I wrote this when intersectBase can never be 2.

It totally could be, if someone were to write any code that called it in that way, it just never is.

> It would be safer to change the type to int8_t.

Done.